modelon/impact/client/entities/experiment.py

```diff
@@ -22,6 +22,52 @@
 
 logger = logging.getLogger(__name__)
 
+ScalarValue = Union[float, int, str]
+
+
+@enum.unique
+class _TrajectoryResponseFormat(enum.Enum):
+    V1 = "application/vnd.impact.trajectories.v1+json"
+    v2 = "application/vnd.impact.trajectories.v2+json"
+
+
+class ExperimentResultPoint:
+    """Value class with the single time instance data in a experiment."""
+
+    def __init__(self, trajectories: List[Dict[str, Any]], variables: List[str]):
+        self._trajectories = trajectories
+        self._variables = variables
+
+    @property
+    def variables(self) -> List[str]:
+        """List of variables."""
+        return self._variables
+
+    @property
+    def cases(self) -> List[str]:
+        """List of case ids."""
+        return [case["caseId"] for case in self._trajectories]
+
+    def as_lists(self) -> List[Optional[List[Optional[ScalarValue]]]]:
+        """Return a list of results per case.
+
+        None indicates that the case was not run or failed. None value
+        for a variable indicated that the variable is not present in the
+        specific case output.
+
+        Example::
+
+            trajectories = result.as_lists()
+
+        """
+        return [
+            [
+                case["items"][i].get("trajectory")[0] if case["items"][i] else None
+                for i, _ in enumerate(self._variables)
+            ]
+            for case in self._trajectories
+        ]
```

Comment:
This kind of loops worth optimizing as they grow a lot for multi-sims, so if I read it right:
```
 [
            [ value["trajectory"][0] if value else None for value in case_data ]
            for case_data in self._trajectories["items"]
] 
```

Reply:
@iakovn  Hmm trajectories is a list of dicts that looks like this, so that **_self._trajectories["items"]_** wont work right?

```
[
            {
                "caseId": "case_1",
                "items": [
                    {"fixed": True, "trajectory": [1.0]},
                    {"fixed": False, "trajectory": [1.0]},
                ],
            },
            {
                "caseId": "case_2",
                "items": [
                    {"fixed": True, "trajectory": [1.0]},
                    {"fixed": False, "trajectory": [2.0]},
                ],
            },
            {
                "caseId": "case_3",
                "items": [
                    {"fixed": True, "trajectory": [1.0]},
                    None,
                ],
            },
        ]
```

Reply:
I can probably change from  enumerate(self._variables) to range(len())? Also compute len only once to and save it in a variable. Something like this

        variable_list_length = len(self._variables)
        return [
            [
                case["items"][i].get("trajectory")[0] if case["items"][i] else None
                for i in range(variable_list_length)
            ]
            for case in self._trajectories
        ]

Reply:
I'd like to avoid repeated `case["items"][i]` in the inner loop. How about:

 ```
 [
            [ item["trajectory"][0] if item else None for value in case_data["items"] ]
            for case_data in self._trajectories
] 
```
BTW: what happens if a case was not run or failed without generating result? Does backend still return "items"? Alternative is that we need additional "if" for `case_data["items"]` presence

Reply:
Thanks for optimizing it :) I cross checked that "items" is present in the response for a failed compilation too :) Could you please take another look? Thanks!